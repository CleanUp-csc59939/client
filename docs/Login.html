<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Login</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Login</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Login<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Login"><span class="type-signature"></span>new Login<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h4>
    

    



<div class="description">
    <h1>&lt;&lt;&lt;&lt;&lt;&lt;&lt; HEAD
return home page with multiple user events*</h1>
<p>return Login page*</p>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<p>05a6d880b7895c0041fd364f094c89906bdcf35e</p>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_login_login.jsx.html">components/login/login.jsx</a>, <a href="components_login_login.jsx.html#line43">line 43</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Return Login page where user can input their credentials</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~handleLogin"><span class="type-signature">(inner) </span>handleLogin<span class="signature">(e)</span><span class="type-signature"> &rarr; {HTMLBodyElement}</span></h4>
    

    



<div class="description">
    <p>function called after user clicks login</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>e</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>Component props</p>
                <h6>Properties</h6>
                

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>preventDefault</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    
    </tbody>
</table>






    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>form</code></td>
            

            <td class="type">
            
                
<span class="param-type">form</span>


            
            </td>

            

            

            <td class="description last"><p>validate the form</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>AuthService</code></td>
            

            <td class="type">
            
                
<span class="param-type">method</span>


            
            </td>

            

            

            <td class="description last"><p>sends email and password input to the authenication service which communicates with the back end</p></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_login_login.jsx.html">components/login/login.jsx</a>, <a href="components_login_login.jsx.html#line88">line 88</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Returns the HTML code of the Login Form</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">HTMLBodyElement</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Banner.html">Banner</a></li><li><a href="CreateMeetUp.html">CreateMeetUp</a></li><li><a href="CustomSearch.html">CustomSearch</a></li><li><a href="EventPics.html">EventPics</a></li><li><a href="Header.html">Header</a></li><li><a href="Home.html">Home</a></li><li><a href="Login.html">Login</a></li><li><a href="MyMeetups.html">MyMeetups</a></li><li><a href="Profile.html">Profile</a></li><li><a href="ProfilePic.html">ProfilePic</a></li><li><a href="ProtectedRoute.html">ProtectedRoute</a></li><li><a href="SignUp.html">SignUp</a></li><li><a href="SingleEvent.html">SingleEvent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#%253E%253E%253E%253E%253E%253E%253E05a6d880b7895c0041fd364f094c89906bdcf35e">>>>>>>> 05a6d880b7895c0041fd364f094c89906bdcf35e</a></li><li><a href="global.html#authHeader">authHeader</a></li><li><a href="global.html#ConvertDate">ConvertDate</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Tue May 25 2021 02:31:10 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>