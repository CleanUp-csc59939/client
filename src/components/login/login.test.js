import React from 'react';
import { shallow } from 'enzyme';
import Login from './login';

describe('Login', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Login debug />);

    expect(component).toMatchSnapshot();
  });
});
