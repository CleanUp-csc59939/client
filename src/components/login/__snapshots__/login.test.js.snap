// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly in "debug" mode 1`] = `
<div
  className="loginBody"
>
  <div
    className="login-page"
  >
    <div
      className="form"
    >
      <Form(t)
        className="login-form"
        onSubmit={[Function]}
      >
        <Control(s)
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="text"
          validations={
            Array [
              [Function],
              [Function],
            ]
          }
          value=""
        />
        <Control(s)
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          validations={
            Array [
              [Function],
            ]
          }
          value=""
        />
        <button
          disabled={false}
          type="submit"
        >
          <span>
            Login
          </span>
        </button>
        <Button(s)
          style={
            Object {
              "display": "none",
            }
          }
        />
        <p
          className="message"
        >
          Not registered? 
          <a
            href="/signup"
          >
            Create an account
          </a>
        </p>
      </Form(t)>
    </div>
  </div>
</div>
`;
